frontend/lib/norent/letter-builder/know-your-rights.tsx

```diff
@@ -4,49 +4,93 @@ import Page from "../../ui/page";
 import { Link } from "react-router-dom";
 import { AppContext } from "../../app-context";
 import { BackButton } from "../../ui/buttons";
+import { getUSStateChoiceLabels } from "../../../../common-data/us-state-choices";
 import {
-  getUSStateChoiceLabels,
-  USStateChoice,
-} from "../../../../common-data/us-state-choices";
-import { getNorentMetadataForUSState } from "./national-metadata";
+  StatePartnerForBuilderEntry,
+  getNorentMetadataForUSState,
+  assertIsUSState,
+  NorentMetadataForUSState,
+} from "./national-metadata";
 import { OutboundLink } from "../../analytics/google-analytics";
+import { getStatesWithLimitedProtectionsFAQSectionURL } from "../faqs";
+
+const StateWithoutProtectionsContent: React.FC<NorentMetadataForUSState> = (
+  props
+) => {
+  return (
+    <>
+      <p>
+        Unfortunately, we do not currently recommend sending a notice of
+        non-payment to your landlord. Sending a notice could put you at risk.{" "}
+        <Link to={getStatesWithLimitedProtectionsFAQSectionURL()}>
+          Learn more.
+        </Link>
+      </p>
+
+      <div className="content">
+        <p>
+          We’ve partnered with <PartnerLink {...props.partner} /> to provide
+          additional support.
+        </p>
+      </div>
+    </>
+  );
+};
+
+export const PartnerLink: React.FC<StatePartnerForBuilderEntry> = (props) => (
+  <OutboundLink
+    href={props.organizationWebsiteLink}
+    target="_blank"
+    rel="noopener noreferrer"
+  >
+    {props.organizationName}
+  </OutboundLink>
+);
+
+export const StateWithProtectionsContent: React.FC<NorentMetadataForUSState> = (
+  props
+) => (
+  <>
+    <p>{props.lawForBuilder.textOfLegislation}</p>
+
+    <div className="content">
+      <p>
+        We’ve partnered with <PartnerLink {...props.partner} /> to provide
+        additional support once you’ve sent your letter.
+      </p>
+    </div>
+  </>
+);
 
 export const NorentLbKnowYourRights = MiddleProgressStep((props) => {
   const { session } = useContext(AppContext);
-  const state = session.norentScaffolding?.state as USStateChoice;
+  const scf = session.norentScaffolding;
 
-  // Content from national metadata:
-  const legislation = getNorentMetadataForUSState(state)?.lawForBuilder
-    ?.textOfLegislation;
+  if (!scf?.state) {
+    return (
+      <p>
+        Please <Link to={props.prevStep}>go back and choose a state</Link>.
+      </p>
```

Comment:
This is very makeshift and will go away once I implement the whole "you're not supposed to be here!" thing for pages that we don't have enough session info to render meaningful content with.

Reply:
cool cool, good to know